readyplayerme/rpm-unity-sdk-core
# [SDK-254] Feature/add amplitude events

## [SDK-254](https://ready-player-me.atlassian.net/browse/SDK-254)


## Dependencies
```Package
Avatar Loader: branch
```

## Description

-   This PR adds some new amplitude events for analytics
- it also includes some refactoring and the analytics classes




## Changes

#### Added

-   new events for analytics tracking

#### Updated

-   AnalyticsEventLogger refactoring



## How to Test

-   Open a project with this branch of core and enable analytics
- open up amplitude and send some events from unity EG toggling analytics
- track which userIDis yours
- then review and check all the new events are firing for that user 



## Checklist








[SDK-254]: https://ready-player-me.atlassian.net/browse/SDK-254?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
